src/xFactor/Stage.java

```diff
@@ -0,0 +1,87 @@
+package xFactor;
+
+import java.util.ArrayList;
+
+public class Stage {
+	private static int numberOfStages = 8;
+	private static int countInstances;
+	private int stageNumber;
+	private int maxParticipants;
+	private ArrayList<Participant> participantsInStage = new ArrayList<Participant>(maxParticipants);
+	private ArrayList<Participant> qualifiedParticipants = new ArrayList<Participant>();
+	private static ArrayList<Participant> tempArray = new ArrayList<Participant>();
+	
+	public Stage(int stageNum, int max) throws IllegalArgumentException {
+		this.stageNumber = stageNum;
+		countInstances++;
+		initializeInstance(max, stageNum);
+	}
+	
+	public int getStageNumber() {
+		return stageNumber;
+	}
+
+	public int getMaxParticipants() {
+		return maxParticipants;
+	}
+
+	public static int getNumberOfStages() {
+		return numberOfStages;
+	}
+
+	public static void setNumberOfStages(int numberOfStages) {
+		Stage.numberOfStages = numberOfStages;
+	}
+	
+	public ArrayList<Participant> getParticipantsInStage() {
+		return participantsInStage;
+	}
+	
+	public ArrayList<Participant> getQualifiedParticipants() {
+		if(qualifiedParticipants.size() == 1) {
+			System.out.println("The winner of X-Factor is: " + qualifiedParticipants.get(0));
+		}
+		return qualifiedParticipants;
+	}
+	
+	private void initializeInstance(int max, int stageNum) {
+		if(stageNum < 1 || stageNum > numberOfStages) {
+			throw new IllegalArgumentException("The number of stage must be between 1 and 12.");

```

Comment:
you should not hardcode this if you have a numberOfStages, what will happen if someone changes it to 5?